core/src/databases/table.rs

```diff
@@ -409,16 +413,23 @@ impl LocalTable {
                         ))?;
                     }
 
-                    match row.value().keys().find(|key| match key.chars().next() {
-                        Some(c) => c.is_ascii_uppercase(),
-                        None => false,
-                    }) {
-                        Some(key) => Err(anyhow!(
-                            "Row {} has a key '{}' that contains uppercase characters",
-                            row_index,
-                            key
-                        ))?,
-                        None => (),
+                    for header in row.headers.iter() {
+                        if let Some(first_char) = header.chars().next() {
```

Comment:
I ended up reimplementing the uppercase check here since we were already iterating over headers. Should also be more efficient by avoiding `row.value()`, which is very expensive

Reply:
I think the intent was to check for the whole string but this was cargo-culted from https://github.com/dust-tt/dust/blame/fd43a01ea17739232ac2565276ccc965ede023fa/core/src/databases/table.rs#L240-L251 so I presume the question goes to @fontanierh? :)